device_backends/uio/src/UioAccess.cc

```diff
@@ -32,14 +34,13 @@ namespace ChimeraTK {
   }
 
   void UioAccess::open() {
-    UioMMap();
+    _mmio = std::make_unique<MmioAccess>(_deviceFileDescriptor, _deviceMemSize, true);
     _opened = true;
   }
 
   void UioAccess::close() {
     if(!_opened) {
-      UioUnmap();
-      ::close(_deviceFileDescriptor);
+      _mmio.reset();
```

Comment:
Here is probably a behaviour change as you removed the `::close(_deviceFileDescriptor)`. Is that intentional?

Reply:
No, the MmioAccess takes ownership of the file descriptor so resetting the pointer will close the fd and leave it dangling as the old code would.

Reply:
Uh, that's not clear from the code. Can the MmioAccess object even create the _deviceFileDescriptor in the first place so the parent class does not deal with it at all? I guess that would be inline with some of the previous comments...

Reply:
This will need deeper rework since it affects the interrupt handler, I am already looking into it. I think it will be much clearer in the end

Reply:
I would prefer using [DeviceFile](https://github.com/ChimeraTK/DeviceAccess/blob/master/device_backends/xdma/include/DeviceFile.h) for that, instead of dealing with explicit fd's and explicit closing, but looking at https://github.com/ChimeraTK/DeviceAccess/pull/358#discussion_r1312699597 maybe that doesn't fit into the style of the rest of the framework.

Reply:
That comment was not against the DeviceFile object in total, merely a small question mark about a design aspect of DeviceFile...